core/common/src/main/java/alluxio/AbstractClient.java

```diff
@@ -173,6 +179,16 @@ public synchronized void connect() throws IOException, ConnectionFailedException
         afterConnect();
         checkVersion(getClient(), getServiceVersion());
         return;
+      } catch (IOException e) {
+        if (FRAME_SIZE_EXCEPTION_PATTERN.matcher(e.toString()).find()) {
```

Comment:
Oh okay, that works for me (but I guess `getLocalizedMessage` might be more precise?).